### ISSUE TITLE
[no-shadow] False positive on `export enum X` declaration

### ISSUE THREAD

**Issue by AArnott** [typescript-eslint/typescript-eslint#2471]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```
git clone https://github.com/AArnott/pr-autocomplete-app
cd pr-autocomplete-app
git checkout eslintbug_2471
yarn
yarn lint
```

A single `export enum` in the entire compilation, with no imports, such as this:

```TS
export enum MergeMethods3 {
  merge = "merge",
}
```

**Expected Result**

No lint errors

**Actual Result**

```
/home/andrew/git/pr-autocomplete-app/GitHubHook/index.ts
  1:13  error  'MergeMethods3' is already declared in the upper scope  no-shadow
```

**Additional Info**

This regressed in https://github.com/AArnott/pr-autocomplete-app/pull/65 where we upgrade `@typescript-eslint/parser` from 3.10.1 to 4.0.1.

No matter what I name the enum, it rejects it with the same error.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.0.1` |
| `@typescript-eslint/parser`        | `4.0.1` |
| `TypeScript`                       | `4.0.2` |
| `ESLint`                           | `7.8.1` |
| `node`                             | `12.16.1` |

**Comment by bradzacher** [comment]
As per the release notes (https://github.com/typescript-eslint/typescript-eslint/releases/tag/v4.0.0) you need to use `@typescript-eslint/no-shadow`, not the base eslint rule.

See also https://github.com/typescript-eslint/typescript-eslint/issues/2466

**Comment by karladler** [comment]
Fixed it for me:
```
'no-shadow': 'off', // replaced by ts-eslint rule below
 '@typescript-eslint/no-shadow': 'error'
```
